[macOS] fails to build Lua with space in path

I installed `lux-cli` via nix but failed to run `lx lua`. Seems like there is an issue while building lua binary.

```
{
  description = "A very basic dev flake";

  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
    lux.url = "github:lumen-oss/lux";
  };

  outputs = { self, nixpkgs, lux, ... }: let
    system = "aarch64-darwin";
    pkgs = import nixpkgs {
      inherit system;
      overlays = [ lux.overlays.default ];
    };
  in
    {
      devShells.${system}.default = pkgs.mkShell {
        buildInputs = [
          pkgs.lux-cli
          pkgs.lux-lua51
        ];
      };
    };
}
```

```
$ lx version
lux-cli 0.18.0
```

```
$ nix develop

error: builder for '/nix/store/hf7664j1958990mycyn2plxmkb9cgw39-lux-lua-0.18.0.drv' failed with exit code 255;
       last 15 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/jjpcpml8fkfb20yl4z7hmsf4dd7smj8p-source
       > source root is source
       > Running phase: patchPhase
       > Executing configureCargoCommonVars
       > decompressing cargo artifacts from /nix/store/y4aqg8k253f6sgkhzn7hbzsr6b0ammkb-lux-deps-0.1.0/target.tar.zst to target
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > will append /private/tmp/nix-build-lux-lua-0.18.0.drv-0/source/.cargo-home/config.toml with contents of /nix/store/y234jnzlbja6yyhlmrdqk6d6rqidlzch-vendor-cargo-deps/config.toml
       > default configurePhase, nothing to do
       > Running phase: buildPhase
       > +++ command cargo --version
       > cargo 1.89.0 (c24e10642 2025-06-23)
       > error: could not find `Cargo.toml` in `/private/tmp/nix-build-xtask-lua51-0.18.0.drv-0/source` or any parent directory
       > cargo build failed
       For full logs, run:
         nix log /nix/store/hf7664j1958990mycyn2plxmkb9cgw39-lux-lua-0.18.0.drv
error: 1 dependencies of derivation '/nix/store/13rmx052f6jzknyk12cd90jhll8x8wx8-nix-shell-env.drv' failed to build
```

removing `pkgs.lux-lua51` and running `lx lua` under lua project directory (created with `lx new`) also failed:

<details>
<summary>result</summary>

```
$ lx lua # under directory with lux.toml

Error: could not find a Lua installation and failed to build Lua from source:
build failed.

exit status: 2

stdout:
cd src && make generic
make[1]: Entering directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'
make all MYCFLAGS=
make[2]: Entering directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'
gcc -O2 -Wall    -c -o lapi.o lapi.c
make[2]: Leaving directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'
make[1]: Leaving directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'


stderr:
error: tool 'gcc' not found
make[2]: *** [<builtin>: lapi.o] Error 1
make[1]: *** [Makefile:96: generic] Error 2
make: *** [Makefile:56: generic] Error 2


Caused by:
    build failed.

    exit status: 2

    stdout:
    cd src && make generic
    make[1]: Entering directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'
    make all MYCFLAGS=
    make[2]: Entering directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'
    gcc -O2 -Wall    -c -o lapi.o lapi.c
    make[2]: Leaving directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'
    make[1]: Leaving directory '/private/tmp/nix-shell.JaVaXh/.tmpuLv3xJ/src'


    stderr:
    error: tool 'gcc' not found
    make[2]: *** [<builtin>: lapi.o] Error 1
    make[1]: *** [Makefile:96: generic] Error 2
    make: *** [Makefile:56: generic] Error 2


Location:
    /private/tmp/nix-build-lux-cli-0.18.0.drv-0/source/lux-cli/src/build.rs:24:18
```

</details>

After I provided `pkgs.gcc` in `buildInputs`:

<details>
<summary>result</summary>

```
$ lx lua # under directory with lux.toml

Error: could not find a Lua installation and failed to build Lua from source:
install failed.

exit status: 2

stdout:
cd src && mkdir -p /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/bin /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/include /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/lib /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/man/man1 /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/share/lua/5.1 /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/lib/lua/5.1
cd src && install -p -m 0755 lua luac /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/bin


stderr:
install: omitting directory '/Users/boltless/Library/Application'
make: *** [Makefile:63: install] Error 1


Caused by:
    install failed.

    exit status: 2

    stdout:
    cd src && mkdir -p /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/bin /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/include /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/lib /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/man/man1 /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/share/lua/5.1 /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/lib/lua/5.1
    cd src && install -p -m 0755 lua luac /Users/boltless/Library/Application Support/org.neorocks.lux/tree/5.1/.lua/bin


    stderr:
    install: omitting directory '/Users/boltless/Library/Application'
    make: *** [Makefile:63: install] Error 1


Location:
    /private/tmp/nix-build-lux-cli-0.18.0.drv-0/source/lux-cli/src/build.rs:24:18
```

</details>


edit: markdown formatting

--- a/lux-lib/src/operations/build_lua.rs
+++ b/lux-lib/src/operations/build_lua.rs
@@ -189,7 +189,7 @@ async fn do_build_luajit_unix(args: BuildLua<'_>, build_dir: &Path) -> Result<()
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .arg("install")
-        .arg(format!("PREFIX={}", install_dir.display()))
+        .arg(format!(r#"PREFIX="{}""#, install_dir.display()))
         .output()
         .await
     {
@@ -380,7 +380,7 @@ async fn do_build_lua_unix(
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .arg("install")
-        .arg(format!("INSTALL_TOP={}", install_dir.display()))
+        .arg(format!(r#"INSTALL_TOP="{}""#, install_dir.display()))
         .output()
         .await
     {

